vmdb/app/models/ems_refresh/parsers/openstack_infra.rb

```diff
@@ -56,25 +72,35 @@ def load_hosts
         indexed_hosts_ports = {}
         hosts_ports.each { |p|  (indexed_hosts_ports[p.uuid] ||= []) <<  p }
 
+        # Indexed Heat resources, we are interested only in OS::Nova::Server
+        indexed_resources = {}
+        resources.each { |p| indexed_resources[p['physical_resource_id']] = p }
+
         process_collection(hosts, :hosts) do  |host|
-          parse_host(host, indexed_servers, indexed_hosts_ports)
+          parse_host(host, indexed_servers, indexed_hosts_ports, indexed_resources)
         end
       end
 
-      def parse_host(host, indexed_servers, _indexed_hosts_ports)
+      def parse_host(host, indexed_servers, _indexed_hosts_ports, indexed_resources)
         uid = host.uuid
+        hostname = identify_hostname(indexed_resources, host.instance_uuid)
```

Comment:
@fryguy, do you see a problem with this?  We were recently talking about funky hostnames in another context.  I'm not sure storing `unknown hostname (#{purpose})` will work in the `hostname` field.

Also, there's potential for the `hostname` to be nil.  Not sure what impact that will have in other parts.